src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -304,6 +305,12 @@ void DirectoryService::InjectPoWForDSNode(
       LOG_GENERAL(INFO, "Injecting into PoW connections " << rit->second);
     }
 
+    if (m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() >=
+        UPGRADE_TARGET_DS_NUM) {
+      // Remove this node from blacklist if it exists
```

Comment:
Do you think everyone (DS, former DS, and new shard nodes) should remove them from blacklist? If so, doing it here might be easier: https://github.com/Zilliqa/Zilliqa/blob/d5cc251e6b3703dc9c70b8f871a63d6c76ba5593/src/libDirectoryService/DSComposition.cpp#L137

Reply:
Yes. Can do